### PR TITLE
Improve notifications

### PR DESCRIPTION
**Pull request by liviu-timar**
### 🎯 Goal

Refactor notifications to show up-to-date text and icons.

### 🛠 Implementation details

- Added new `getOngoingCallNotification` method in `NotificationHandler` with an `onUpdate` lambda parameter.
- The default implementation of `getOngoingCallNotification` observes the ringing state and number of members/participants and calls `onUpdate` with a refreshed notification that reflects the call state.
- `getOngoingCallNotification` is used in `CallService`. Will also be used in the `Telecom` implementation in the future.
- Various changes to `DefaultNotificationHandler` and `CallService`
- Changed notification titles, desc, icons.
- ❗ Breaking change: the `callDisplayName` parameter in `NotificationHandler#getRingingCallNotification()` is now nullable (`String` -> `String?`).

### 🎨 UI Changes

image

### 🧪 Testing

- Go through the entire ringing flow and watch all notifications: incoming, outgoing, ongoing.
- Start a group call and add/remove participants: incoming, outgoing, ongoing.

### CODE REVIEW

**Comment by liviu-timar** [comment]
Found a bug. Pressing on the ongoing call `Hang up` button does not work on the caller device, only on receiver.

**Review by liviu-timar — COMMENTED** [review]
Found a bug. Pressing on the ongoing call `Hang Up` button does not work on the caller device, only on receiver. Fixing it.

File: stream-video-android-core/src/main/kotlin/io/getstream/video/android/core/notifications/DefaultNotificationHandler.kt [hunk 1]
```diff
@@ -332,7 +300,11 @@ public open class DefaultNotificationHandler(
             callId,
             notificationId,
         )
-        val endCallIntent = intentResolver.searchEndCallPendingIntent(callId = callId)
+        val hangUpIntent = if (isOutgoingCall) {
```
Comment (liviu-timar) [inline stream-video-android-core/src/main/kotlin/io/getstream/video/android/core/notifications/DefaultNotificationHandler.kt:310]: Bug here, hang-up not working for outgoing device. Looking into it.

File: stream-video-android-core/src/main/kotlin/io/getstream/video/android/core/notifications/NotificationHandler.kt [hunk 2]
```diff
@@ -18,22 +18,36 @@ package io.getstream.video.android.core.notifications
...
+        callId: StreamCallId,
+        callDisplayName: String?,
+        isOutgoingCall: Boolean = false,
+        remoteParticipantCount: Int = 0,
+    ): Notification?
     fun getRingingCallNotification(
         ringingState: RingingState,
         callId: StreamCallId,
-        callDisplayName: String,
+        callDisplayName: String?,
```
Comment (aleksandar-apostolov) [inline stream-video-android-core/src/main/kotlin/io/getstream/video/android/core/notifications/NotificationHandler.kt]: can we have a default `null` value?

Comment (liviu-timar) [reply stream-video-android-core/src/main/kotlin/io/getstream/video/android/core/notifications/NotificationHandler.kt]: Yes, would be good. Also did it for `getOngoingCallNotification`.

**Review by aleksandar-apostolov — APPROVED** [review]

**Pull request merged**